torch/csrc/jit/script/compiler.cpp

```diff
@@ -1368,9 +1368,7 @@ struct to_ir {
 
   // Emit nodes for augmented assignments like `+=`
   void emitAugAssignment(const AugAssign& stmt) {
-    auto lhs = Var(stmt.lhs());
-    auto lhsValue = environment_stack->getSugaredVar(lhs.name())
-                        ->asValue(lhs.range(), method);
+    auto lhsValue = emitExpr(stmt.lhs());
```

Comment:
In the else branch below there is a cast to `Var` so `a[i] += 1` will fail on a list of integers. Beware desugaring to `a[i] = a[i] + 1` is invalid because both `i`  and `a` may actually be some expression with side effects and cannot be repeated. Subscripted augmented assigns will need to ensure they only compute `a` and and `i` only once.